pkg/collector/harbor_events.go

```diff
@@ -0,0 +1,508 @@
+
+	occurrences := make([]*grafeas.Occurrence, 0)
+	for i := range resources {
+		scanOverview := resources[i].ScanOverview["application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0"].(map[string]interface{})
+		if scanOverview["scan_status"].(string) == "Success" {
+			status = discovery.Discovered_FINISHED_SUCCESS
+			vulnerabilityDetails = t.getVulnerabilityDetails(scanOverview["severity"].(string))
+		} else if scanOverview["scan_status"].(string) == "Error" {
+			status = discovery.Discovered_FINISHED_FAILED
+		}
+
+		discoveryDetails := &grafeas.Occurrence_Discovered{
+			Discovered: &discovery.Details{
+				Discovered: &discovery.Discovered{
+					AnalysisStatus: status,
+				},
+			},
+		}
+
+		o := newHarborImageScanOccurrence(resources[i], t.project)
+		o.Details = discoveryDetails
+		occurrences = append(occurrences, o)
+		if scanOverview["scan_status"].(string) == "Success" {
+			o = newHarborImageScanOccurrence(resources[i], t.project)
+			o.Details = vulnerabilityDetails
+			occurrences = append(occurrences, o)
+		}
+
+	}
+	return occurrences
+}
+
+func newHarborImageScanOccurrence(resource *Resource, projectName string) *grafeas.Occurrence {
+	o := &grafeas.Occurrence{
+		Resource: &grafeas.Resource{
+			Uri: HarborOccurrenceResourceURI(resource.ResourceURL, resource.Digest),
+		},
+		NoteName: HarborOccurrenceNote(projectName),
+	}
+	return o
+}
+
+func HarborOccurrenceResourceURI(url, digest string) string {
+	return fmt.Sprintf("%s@%s", url, digest)
+}
+
+func HarborOccurrenceNote(projectName string) string {
+	return fmt.Sprintf("projects/%s/notes/%s", "rode", projectName)
+}
+
+func (t *HarborEventCollector) getVulnerabilityDetails(severity string) *grafeas.Occurrence_Vulnerability {
+	vulnerabilitySeverity := t.getVulnerabilitySeverity(severity)
+	vulnerabilityDetails := &grafeas.Occurrence_Vulnerability{
+		Vulnerability: &vulnerability.Details{
+			Severity: vulnerabilitySeverity,
+			PackageIssue: []*vulnerability.PackageIssue{
+				{
+					AffectedLocation: &vulnerability.VulnerabilityLocation{
+						CpeUri:  "TODO",
+						Package: "TODO",
+						Version: &packag.Version{
+							Kind: packag.Version_NORMAL,
+							Name: "TODO",
+						},
+					},
+				},
+			},
+		},
+	}
+	return vulnerabilityDetails
+}
+
+func (t *HarborEventCollector) getVulnerabilitySeverity(v string) vulnerability.Severity {
+	switch v {
+	case HarborSeverityCritical:
+		return vulnerability.Severity_CRITICAL
+	case HarborSeverityHigh:
+		return vulnerability.Severity_HIGH
+	case HarborSeverityMedium:
+		return vulnerability.Severity_MEDIUM
+	case HarborSeverityLow:
+		return vulnerability.Severity_LOW
+	case HarborSeverityNegligible:
+		return vulnerability.Severity_MINIMAL
+	case HarborSeverityNone:
+		return vulnerability.Severity_MINIMAL
+	default:
+		return vulnerability.Severity_SEVERITY_UNSPECIFIED
+	}
+}
+
+// Assumes Harbor admin creds are deployed in the same namespace as the Collector CR
```

Comment:
I don't think this is a safe assumption to make.  When the user specifies the harbor secret, they should probably use the format `${namespace}/${secretName}`.  We can parse that to figure out what namespace to use.